evo_researcher/benchmark/benchmark.py

```diff
@@ -70,18 +95,26 @@ def get_prediction_result(market: Market):
                             completion_tokens=cb.completion_tokens,
                         )
                     prediction.cost = cb.total_cost
-                return prediction
+                return market.question, prediction
 
             # Run agents in parallel
             with concurrent.futures.ThreadPoolExecutor(
                 max_workers=agent.max_workers
             ) as executor:
                 future_to_market = {
                     executor.submit(get_prediction_result, market): market
-                    for market in self.markets
+                    for market in markets_to_run
                 }
                 for future in concurrent.futures.as_completed(future_to_market):
-                    self.add_prediction(agent=agent, prediction=future.result())
+                    market_question, prediction = future.result()
+                    self.add_prediction(
+                        agent=agent,
+                        prediction=prediction,
+                        market_question=market_question,
+                    )
+
+        if self.cache_path:
+            PredictionsCache(predictions=self.predictions).save(self.cache_path)
```

Comment:
Could this be executed after each agent's predictions? So if one agent fails, we don't need to rerun everything.

Reply:
Yes, I think that's a great idea. Not super straight forward with this implementation, as the predictions are made in separate threads atm, but would like to do this as a follow-up

Reply:
In my stolen branch, I just indented it 2 times:

```python
for future in concurrent.futures.as_completed(future_to_market):
                    market_question, prediction = future.result()
                    self.add_prediction(
                        agent=agent,
                        prediction=prediction,
                        market_question=market_question,
                    )
                    if self.cache_path:
                        PredictionsCache(predictions=self.predictions).save(self.cache_path)
```

Predictions are running in threads, but this saving is running sequentially, as the threads are completed, isn't it?

Reply:
Ah nice, that works!